pkg/resources/resource.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package resources
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/workload"
+)
+
+type FlavorResource struct {
+	Flavor   kueue.ResourceFlavorReference
+	Resource corev1.ResourceName
+}
+
+type FlavorResourceQuantities map[kueue.ResourceFlavorReference]workload.Requests
+type FlavorResourceQuantitiesFlat map[FlavorResource]int64
+
+func Unflatten(f FlavorResourceQuantitiesFlat) FlavorResourceQuantities {
```

Comment:
Is there a reason why it's not a FlavorResourceQuantitiesFlat's method?

Reply:
Nope. I will update to a method.